src/mots/directory.py

```diff
@@ -0,0 +1,216 @@
+        for module in self.modules:
+            self.modules_by_machine_name[module.machine_name] = module
+
+        self.index = None
+        self.people = None
+
+    def load(self, full_paths: bool = False, query_bmo=True):
+        """Load all paths in each module and put them in index."""
+        self.index = defaultdict(list)
+
+        if full_paths:
+            # Make sure all existing paths have an entry in the index.
+            logger.debug("Loading paths into index...")
+            for path in self.repo_path.glob("**/*"):
+                if path.relative_to(self.repo_path).parts[0] in (".hg", ".git"):
+                    continue
+                self.index[path] = list()
+            logger.debug(f"{len(self.index)} paths loaded.")
+
+        for module in self.modules:
+            if module.submodules:
+                # Start with more specific submodule definitions if present.
+                for submodule in module.submodules:
+                    logger.debug(f"Updating index for {submodule.machine_name}...")
+                    for path in submodule.calculate_paths():
+                        self.index[path].append(submodule)
+            logger.debug(f"Updating index for {module.machine_name}...")
+            # Add broader module definitions.
+            for path in module.calculate_paths():
+                self.index[path].append(module)
+
+        # Filter out modules that specifically exclude paths defined in other modules.
+        for path in self.index.keys():
+            modules = self.index[path]
+            if len(modules) > 1:
+                self.index[path] = [m for m in modules if not m.exclude_module_paths]
+        self.index = dict(self.index)
+
+        # Load people directory
+        people = list(self.config_handle.config["people"])
+        bmo_data = _get_bmo_data(people) if query_bmo else {}
+        self.people = People(people, bmo_data)
+        if self.people.serialized != list(self.config_handle.config["people"]):
+            logger.debug("People directory modified, updating configuration...")
+            self.config_handle.config["people"] = self.people.serialized
+
+    def query(self, *paths: str) -> QueryResult:
+        """Query given paths and return a list of corresponding modules."""
+        result = {}
+        rejected = []
+        for path in paths:
+            if not (self.repo_path / path).exists():
+                logger.warning(f"Path {path} does not exist, skipping.")
+                rejected.append(path)
+                continue
+            result[path] = self.index.get(self.repo_path / path, list())
+        logger.debug(f"Query {paths} resolved to {result}.")
+
+        return QueryResult(result, rejected)
+
+
+class QueryResult:
+    """Helper class to simplify query result interpretation."""
+
+    paths: list = None
+    path_map: dict = None
+    rejected_paths: list = None
+    modules: list = None
+    owners: list = None
+    peers: list = None
```

Comment:
I don't think that these variables are used.
I'm guessing that they're here to imply typings for `data_keys`, but they might be better represented as docs - currently they're misleading due to not being written/accessed.

Also, are the `None` values necessary?